include/swift/Reflection/ReflectionContext.h

```diff
@@ -1377,8 +1377,15 @@ class ReflectionContext
       return std::string("failure reading allocation pool contents");
     auto Pool = reinterpret_cast<const PoolRange *>(PoolBytes.get());
 
+    // Limit how many iterations of this loop we'll do, to avoid potential
+    // infinite loops when reading bad data. Limit to 1 million iterations. In
+    // normal operation, each pool allocation is 16kB, so that would be ~16GB of
+    // metadata which is far more than any normal program should have.
```

Comment:
Started pondering the merits of this vs the tortoise-and-hare. The latter would detect cycles faster in the case of a short cycle. This is simple though, and 16GB of metadata would indeed be crazy.

Reply:
I had similar thoughts. I don't think we care too much about being fast, since this is kind of an emergency situation. We mostly just don't want to crash or hang forever.

Tortoise and hare would also increase our memory reads by 50% in the non-pathological case, which would be mildly unfortunate, although not a huge deal. We could even go for a hybrid model where we only start checking for cycles after N iterations. But I think this way is a good start.